### PR TITLE
refactor(jscs): port jscs to eslint

### LINKED ISSUE #3669: JSCS to be ported to ESLint?
According to this comment: https://github.com/jscs-dev/node-jscs/issues/1511#issuecomment-210822517, JSCS is no longer in production and is being combined with ESLint. So is it time to revisit this question: https://github.com/mozilla/fxa-content-server/issues/2598, remove JSCS and map the rules to the equivalent ESLint rules?

@pdehaan 
@vladikoff

### PR DESCRIPTION
**Pull request by TDA**
Fixes #3669

So more questions: Is this change to be done here, or on `eslint-config-fxa`?
JSCS seems to have not caught these all these ~~years~~ decades, and I vote for it to be removed.

EDIT: Manually tested *every* rule I removed, and eslint catches all of them, woooohoooo! 🎉 

@pdehaan 
@shane-tomlinson 
@vladikoff

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -7,6 +7,14 @@
     "handle-callback-err": 0,
     "no-console": 0,
     "sorting/sort-object-props": [ 2, { ignoreCase: true, ignoreMethods: true, ignoreObjectsWithMethods: true, ignorePrivate: true } ],
-    "strict": [2, "never"]
+    "strict": [2, "never"],
```
Comment (TDA) [inline .eslintrc]: inb4 @shane-tomlinson asks to sort

File: .eslintrc [hunk 2]
```diff
@@ -7,6 +7,14 @@
...
+    "brace-style": 2,
+    "no-multi-str": 2,
+    "key-spacing": [2, {"beforeColon": false, "afterColon": true}],
+    "space-unary-ops": [2, {"nonwords": false, "overrides": {"~": true, "!": true}}],
+    "max-len": [2, 160],
+    "space-infix-ops": 2,
+    "keyword-spacing": 2,
+    "linebreak-style": [2, "unix"]
```
Comment (pdehaan) [inline .eslintrc]: But bonus points if you sort lines 7-18...

Comment (TDA) [inline .eslintrc]: Ha! beat you to it!!! @pdehaan

File: app/scripts/models/account.js [hunk 3]
```diff
@@ -120,7 +120,7 @@ define(function (require, exports, module) {
               self._invalidateSession();
             }
             // Ignore other errors; we'll just fetch again when needed
-          }); /* HACK: See eslint/eslint#1801 */ // eslint-disable-line indent
```
Comment (vladikoff) [inline app/scripts/models/account.js]: 👍

**Comment by vladikoff** [comment]
@TDA need to remove JSCS from devDeps

**Comment by vladikoff** [comment]
Task needs to be removed: https://github.com/mozilla/fxa-content-server/blob/master/grunttasks/jscs.js and wherever it is called too

**Comment by vladikoff** [comment]
update docs https://github.com/mozilla/fxa-content-server/blob/3f52b0ed382ab479154eef327b16aed70cef849c/README.md and https://github.com/mozilla/fxa-content-server/blob/3b40424c1ccbce9dc142f6c0302811c422c697fa/CONTRIBUTING.md

**Comment by TDA** [comment]
@vladikoff Done. Filing a follow-up issue for JSDoc's (after discussion with @pdehaan )

**Comment by pdehaan** [comment]
... Only because I was namechecked:

1. Remove .jscsrc reference from [./grunttasks/jsonlint.js:20](https://github.com/mozilla/fxa-content-server/blob/d616a4baaefeed7e175d9d0b7d0bc4d9ff55e965/grunttasks/jsonlint.js#L20)
1. Remove **jscs-jsdoc** from the [./package.json:101](https://github.com/mozilla/fxa-content-server/blob/d616a4baaefeed7e175d9d0b7d0bc4d9ff55e965/package.json#L101)
1. Update npm-shrinkwrap.json?

```sh
$ git grep jscs

CHANGELOG.md:* **jscs:** allow the 'other' quote mark to be used, but only to avoid having to escape ([b4fc1fda](https://github.com/mozilla/fxa-content-server/commit/b4fc1fdaef199922ed2518071b7ae7008095b0af))
grunttasks/jsonlint.js:        '.jscsrc'
npm-shrinkwrap.json:    "grunt-jscs": {
npm-shrinkwrap.json:      "from": "grunt-jscs@2.8.0",
npm-shrinkwrap.json:      "resolved": "https://registry.npmjs.org/grunt-jscs/-/grunt-jscs-2.8.0.tgz",
npm-shrinkwrap.json:        "jscs": {
npm-shrinkwrap.json:          "from": "jscs@>=2.11.0 <2.12.0",
npm-shrinkwrap.json:          "resolved": "https://registry.npmjs.org/jscs/-/jscs-2.11.0.tgz",
npm-shrinkwrap.json:            "babel-jscs": {
npm-shrinkwrap.json:              "from": "babel-jscs@>=2.0.0 <3.0.0",
npm-shrinkwrap.json:              "resolved": "https://registry.npmjs.org/babel-jscs/-/babel-jscs-2.0.5.tgz",
npm-shrinkwrap.json:            "jscs-preset-wikimedia": {
npm-shrinkwrap.json:              "from": "jscs-preset-wikimedia@>=1.0.0 <1.1.0",
npm-shrinkwrap.json:              "resolved": "https://registry.npmjs.org/jscs-preset-wikimedia/-/jscs-preset-wikimedia-1.0.0.tgz"
npm-shrinkwrap.json:    "jscs-jsdoc": {
npm-shrinkwrap.json:      "from": "jscs-jsdoc@1.3.2",
npm-shrinkwrap.json:      "resolved": "https://registry.npmjs.org/jscs-jsdoc/-/jscs-jsdoc-1.3.2.tgz",
package.json:    "jscs-jsdoc": "1.3.2",
```

**Comment by TDA** [comment]
@pdehaan 
1. Done
2. Done
3. Done, using npmshrink:prod. 
Fixed the jsonlint error as well.

**Comment by vladikoff** [comment]
Excellent, thanks!!

**Pull request merged**